internal/expr/functions/definition.go

```diff
@@ -0,0 +1,75 @@
+package functions
+
+import (
+	"strings"
+
+	"github.com/genjidb/genji/internal/expr"
+	"github.com/genjidb/genji/internal/stringutil"
+)
+
+// A Definition transforms a list of expressions into a Function.
+type Definition interface {
+	Name() string
+	String() string
+	Function(...expr.Expr) (expr.Function, error)
+	Arity() int
+}
+
+// A Definitions table holds a map of definitions, indexed by their names.
+type DefinitionsTable map[string]Definition
+
+// PackagesTable represents a table of SQL functions grouped by their packages
+type PackagesTable map[string]DefinitionsTable
```

Comment:
Ah, I didn't feel this way because it reminded me of functions tables (vtables and others), but it makes sens in the context of a database to avoid it, especially since it's a bit far fetched. 

Duly noted for the conventions 👍